src/random_test.cpp

```diff
@@ -283,34 +322,35 @@ int sum_of_all_server_options() {
   return total;
 }
 
-/* pick some algorith */
+/* pick some algorithm */
 inline static std::string pick_algorithm_lock() {
-  /* pick algorith for current sql */
-  static auto lock = opt_string(LOCK);
-  static auto algorithm = opt_string(ALGORITHM);
-  std::string locks[] = {"DEFAULT", "EXCLUSIVE", "SHARED", "NONE"};
-  std::string algorithms[] = {"INPLACE", "COPY", "INSTANT", "DEFAULT"};
+
   std::string current_lock;
   std::string current_algo;
-  if (lock.compare("all") == 0 && algorithm.compare("all") == 0) {
-    auto lock_index = rand_int(3);
-    auto algo_index = rand_int(3);
-    /* lock=none;algo=inplace|copy not supported */
-    if (lock_index == 3 && algo_index < 2)
-      lock_index = 0;
-    current_lock = locks[lock_index];
-    current_algo = algorithms[algo_index];
-  } else if (lock.compare("all") == 0) {
-    auto lock_index = rand_int(3);
-    current_lock = locks[lock_index];
-    current_algo = algorithm;
-  } else if (algorithm.compare("all") == 0) {
-    auto algo_index = rand_int(3);
-    current_algo = algorithms[algo_index];
-    current_lock = lock;
-  } else {
-    current_lock = lock;
-    current_algo = algorithm;
+
+  current_lock = locks[rand_int(locks.size() - 1)];
+  current_algo = algorithms[rand_int(algorithms.size() - 1)];
+
+  /*
+  Support Matrix	LOCK=DEFAULT	LOCK=EXCLUSIVE	LOCK=NONE	LOCK=SHARED
+  ALGORITHM=INPLACE	Supported	Supported	Supported	Supported
+  ALGORITHM=COPY	Supported	Supported	Not Supported	Supported
+  ALGORITHM=INSTANT	Supported	Not Supported	Not Supported	Not Supported
+  ALGORITHM=DEFAULT	Supported	Supported	Supported	Supported
+*/
+
+  /* If current_algo=INPLACE|DEFAULT, do nothing, since all lock types supported */
+  if (current_algo == "INSTANT")
+    current_lock = "DEFAULT";
+  else if (current_algo == "COPY") {
```

Comment:
we  can set curr_lock=default if current_algo == copy and lock == none.

Reply:
curr_lock can be default, exclusive and shared anything, if current_algo ==copy 
so we cannot hardcode a single value. Hence, I was trying to use probability of randomly picking one.

Reply:
@mohitj1988 better not to add and remove from global variable. It is multi threaded tool. You need to do it under  mutex protection or you might end of get segment error.

Reply:
```
if( current_algo == "copy" && current_lock == "none")
  current_lock = "Default"
```
this way lock=exclusive would be tested whenever  current_lock=exclusive would be picked. 
so this much is enough

Reply:
I want the test coverage to be not missed. In the current implementation also, I see lot of scenarios being missed and also lot of incorrect DDL combinations are generated.
I have tested this patch with all possible combinations thoroughly, it works perfectly fine.

The way you are suggesting is simply hard-coding it, we are not utilisation the randomness and the purpose the vector is created.

Can you please share the test scenario, when we will get segmenr error. I believe I have covered all possible scenarios. Even the scenario when a user simply passes --alter-lock NONE. I have taken care if the vector is empty

Reply:
let me know if I should use mutexes ?

Reply:
which test coverage is missed?  
this hardcoding is just to avoid incorrect DDL combination and use default instead.

you are not seeing crash because We missed to add static. Static are shared resource in multi thread.  So just add static and you migh see fragmentation error.

Reply:
It might generate few incorrect DDL. 
Like instant is suppose to use with add column and not with add index. But we can tolerate some error in pstress because of this simple design or we can improve this in future release for now this is enough.